core-bundle/src/Resources/contao/modules/ModuleNavigation.php

```diff
@@ -69,22 +69,10 @@ protected function compile()
 			$level = ($this->levelOffset > 0) ? $this->levelOffset : 0;
 		}
 
-		$lang = null;
-		$host = null;
-
-		// Overwrite the domain and language if the reference page belongs to a different root page (see #3765)
-		if ($this->defineRoot && $this->rootPage > 0)
-		{
-			$objRootPage = PageModel::findWithDetails($this->rootPage);
-
-			$lang = $objRootPage->rootLanguage;
-			$host = $objRootPage->domain;
-		}
-
```

Comment:
_Note:_ this code was outdated since a while now, since `PageModel::getFrontendUrl` will already generate an URL with the correct domain and language. That's why I removed it and also deprecated the corresponding method parameters of `Module::renderNavigation`.